src/containers/WelcomePage/WelcomePage.jsx

```diff
@@ -26,18 +27,50 @@ import FrontpageSubjects from './FrontpageSubjects';
 import { FILM_PAGE_PATH } from '../../constants';
 import SocialMediaMetadata from '../../components/SocialMediaMetadata';
 import config from '../../config';
+import handleError from '../../util/handleError';
+import { frontpageSearch } from '../../queries';
+import {
+  RESOURCE_TYPE_SUBJECT_MATERIAL,
+  RESOURCE_TYPE_TASKS_AND_ACTIVITIES,
+  RESOURCE_TYPE_LEARNING_PATH,
+  FRONTPAGE_CATEGORIES,
+} from '../../constants';
+
+// making a list from subjects without id
+const topicsNotInNDLA = [];
+FRONTPAGE_CATEGORIES.categories.forEach(category => {
+  if (category && category.subjects) {
+    category.subjects.forEach(subject => {
+      if (subject && !subject.id) {
+        topicsNotInNDLA.push(subject.name);
+      }
+    });
+  }
+});
 
 export class WelcomePage extends Component {
   constructor() {
     super();
     this.state = {
       query: '',
+      inputHasFocus: false,
     };
   }
 
-  onSearchFieldChange = evt => {
-    evt.preventDefault();
-    this.setState({ query: evt.target.value });
+  onSearchInputFocus = () => {
+    this.setState({
+      inputHasFocus: true,
+    });
+  };
+
+  onSearchDeactiveFocusTrap = () => {
+    this.setState({
+      inputHasFocus: false,
+    });
+  };
+
+  onSearchFieldChange = query => {
```

Comment:
Du bør debounce denne funksjonen slik at det ikke gjøres et kall for hvert tastetrykk. Tror det er gjort i masthead-search